homeassistant/components/tuya/__init__.py

```diff
@@ -50,23 +50,23 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
 
 
 async def _init_tuya_sdk(hass: HomeAssistant, entry: ConfigEntry) -> bool:
-    project_type = ProjectType(entry.data[CONF_PROJECT_TYPE])
+    auth_type = AuthType(entry.data[CONF_PROJECT_TYPE])
```

Comment:
Ideally, we should change `CONF_PROJECT_TYPE` to be `CONF_AUTH_TYPE`.
However, we are in beta already so not entirely sure how to handle that.

Options:

1. Just leave it as is (`CONF_PROJECT_TYPE`).
2. Rename it and beta users need to re-add the integration.
3. Rename it and provide migration of the config entry.

I feel like I want to correct this before hitting stable, so we don't have to make a mess later.

Reply:
Option 3 seems sensible?

Reply:
I think that 2 is still acceptable, but definitely not 1

Reply:
Went for option 3, we can remove the migration after a couple of releases.